### PR TITLE
Fix build for linux on travis

### PR DESCRIPTION
**Pull request by slurdge**
And move to xenial

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +19,7 @@ before_install:
 ## Note: all dependencies on MacOS should be resolved by the brew install command
   if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then 
     sudo apt-get update -qq;
-    sudo apt-get install -y gcc-arm-none-eabi;
+    sudo apt-get install -y gcc-arm-none-eabi libnewlib-arm-none-eabi libnewlib-dev binutils-arm-none-eabi;
```
Comment (doegox) [inline .travis.yml]: Current ubuntu-alike instructions is to install
* gcc-arm-none-eabi libstdc++-arm-none-eabi-newlib

while here you install
* gcc-arm-none-eabi libnewlib-arm-none-eabi libnewlib-dev binutils-arm-none-eabi

I'd like to see both aligned. 
libnewlib-arm-none-eabi are libnewlib-dev dependencies of libstdc++-arm-none-eabi-newlib and binutils-arm-none-eabi a recommendation of libstdc++-arm-none-eabi-newlib but maybe this means libstdc++-arm-none-eabi-newlib itself wasn't needed. We should test it a bit further

Comment (slurdge) [reply .travis.yml]: I think stdc++ is not needed. there is no c++ code AFAIK.
I just took the docker xenial and added deps until it compiled. Strangely the different ubuntu flavors do not package the same way arm-non-eabi.

Comment (doegox) [reply .travis.yml]: I tested on a docker ubuntu:xenial
* you can skip binutils-arm-none-eabi, it's a dependency of gcc-arm-none-eabi
* it seems libstdc++-arm-none-eabi-newlib wasn't required.
* but it seems also that libnewlib-arm-none-eabi is not required, only libnewlib-dev

So ``apt-get install gcc-arm-none-eabi libnewlib-dev``
would suffice apparently
I tried removing libnewlib-arm-none-eabi and libstdc++-arm-none-eabi-newlib on my Debian and compile/flash/use runs fine

**Comment by iceman1001** [comment]
Yeah,  I was curious to the new dependecies...
I guess xenial is ubuntu 16.04.  So go go go

**Comment by slurdge** [comment]
Done. OSX still compiling but it's good for linux. We can enable travis and maybe even publish artifacts when we tag a release :)

**Comment by doegox** [comment]
thanks ! @iceman1001 you can enable travis on this repo :)

**Comment by iceman1001** [comment]
Since I am not a CI expert.  How do I enable travis CI?  Thought it needed a seperate branch etc?

**Pull request merged**